src/shared/sources/_lib/validate-source.js

```diff
@@ -88,8 +88,19 @@ function validateSource (source) {
         ))
       }
 
+      requirement(url || paginated, datedError('Require url or paginated'))
+      requirement(([ url, paginated ].filter(s => s).length === 1), datedError('Require url or paginated, not both'))
```

Comment:
Would not `url && paginated` be easier?

Reply:
File this one under "wtf what I thinking."

Reply:
Changed to `requirement(!(url && paginated), datedError('Crawler must have either url or paginated, but not both'))`

Reply:
I _think_ at the time I was imagining a few different options here, rather than a reusable method like there is now.  But who knows ... past me makes odd decisions.